Client/Application/AppDelegate.swift

```diff
@@ -96,6 +108,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
                     }
                 }
             }
+        } else if let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) where components.scheme == "firefox" && components.host == "x-callback-url" {
```

Comment:
Let's clean this up, not cargo-cult it.

```
let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false)
if components.scheme != "firefox" {
    return false
}
switch components.host {
case "firefox":
…
case "x-…"
}
```

You get the idea. Too much nesting, too confused.